packages/fxa-auth-server/lib/oauthdb/index.js

```diff
@@ -99,7 +105,56 @@ module.exports = (log, config) => {
 
     async grantTokensFromAuthorizationCode(oauthParams) {
       try {
-        return await api.grantTokensFromAuthorizationCode(oauthParams);
+        const codeResponse = await api.grantTokensFromAuthorizationCode(
+          oauthParams
+        );
+        const scopeSet = ScopeSet.fromString(codeResponse.scope);
+
+        if (scopeSet.contains(OAUTH_SCOPE_SESSION_TOKEN)) {
+          const origSessionToken = await db.sessionToken(
+            codeResponse.session_token
+          );
+          // Copy verification state of the token, but generate
+          // independent verification codes.
+          let newVerificationState = {};
```

Comment:
@rfk this is from the /duplicate route, we can possibly extract this into a helper

Reply:
I wonder if a `duplicate` method on the `SessionToken` class would suffice for this?  I am so used to things in this codebase not being "proper" classes that I forget that our tokens do actually have that level of abstraction already, and are smart objects rather than just bags of properties.